packages/@aws-cdk/cdk/lib/stack.ts

```diff
@@ -33,16 +32,16 @@ export interface StackProps {
   readonly namingScheme?: IAddressingScheme;
 
   /**
-   * Indicates if this stack is hidden, which means that it won't be
-   * automatically deployed when running `cdk deploy` without arguments.
+   * Indicates if this stack is an entrypoint of your app, which means that it
+   * will be automatically deployed when running `cdk deploy` without arguments.
    *
-   * Setting this to `true` is useful when you have a stack in your CDK app
+   * Setting this to `false` is useful when you have a stack in your CDK app
    * which you don't want to deploy using the CDK toolkit. For example, because
    * you're planning on deploying it through a deployment pipeline.
    *
-   * @default false
+   * @default true
    */
-  readonly hidden?: boolean;
+  readonly entrypoint?: boolean;
```

Comment:
:( Colder (imo)

Reply:
The mental model is that a cloud assembly has a bunch of entrypoints. This is modeled now as an array of artifact IDs at the assembly manifest level. @RomainMuller suggested that we have a single entrypoints section in the assembly
Manifest that can potentially also inform about ordering

Reply:
I see what you're saying, but I don't know that "entrypoint" is the metaphor I would go for though. Entrypoint to me evokes some sort of activity, which I'm not sure this qualifies for.

* top-level?
* standalone?
* target?
* root?

Agree on array of identifiers, don't agree on ordering implications in that array. I feel ordering should be derived from dependency arrows in the dependency graph of artifacts.

Reply:
A different perspective to consider this from: in the future we would only want this value set for dev stacks, right? Most stacks should be CI/CD stacks, which would have this set to false.

In fact, I feel like adding a stack to a pipeline should automatically disable the "autodeploy" feature.

Reply:
* implicit?

Reply:
Removed this feature for now and will add requirement of an argument to "cdk deploy": https://github.com/awslabs/aws-cdk/issues/2750